# Práctica 1.1 — Probabilidad y Estadística
## Análisis de la evolución poblacional mundial

**Alumno:** Luis David Santoyo Romero

En esta práctica se trabaja con datos reales de población obtenidos de **Our World in Data (OWID)**, cuya fuente original es **United Nations, World Population Prospects (2024)**.

El propósito es familiarizarse con un conjunto de datos real y comprobar que el entorno de trabajo (Python, Pandas, NumPy, Jupyter) funciona correctamente.

## Objetivo

Utilizar un conjunto de datos real para comprobar el funcionamiento del entorno de análisis de datos y aplicar medidas estadísticas descriptivas básicas utilizando Pandas y NumPy.

## Actividad 1 — Identificación de la fuente

**Propósito:** reconocer la procedencia del conjunto de datos antes de utilizarlo y documentar correctamente su fuente.

### Cita de la fuente

Los datos utilizados en esta práctica provienen de **Our World in Data (OWID)**.

La fuente original es **United Nations, World Population Prospects (2024)**.

**Cita:**

> UN, World Population Prospects (2024) – processed by Our World in Data. "Annual change in population – UN WPP" [dataset]. United Nations, "World Population Prospects"; United Nations, "World Population Prospects - Interim Update" [original data].

### Preguntas

1. **¿Quién produjo originalmente los datos?**
   La Organización de las Naciones Unidas (UN), a través de su publicación *World Population Prospects (2024)*.

2. **¿Qué organización los procesa para su utilización en Our World in Data?**
   Our World in Data (OWID) es quien recopila, estandariza y procesa los datos originales de la ONU para publicarlos en su plataforma.

3. **¿Qué representa la variable que analizarás?**
   La variable `Annual change in population` representa el cambio neto anual de población (nacimientos, defunciones y migración combinados), calculado como la diferencia entre la población al 1 de julio de dos años consecutivos, expresada en número de personas.

## Actividad 2 — Carga y exploración de datos

**Propósito:** comprobar que el entorno de Python, Pandas y Jupyter funciona correctamente y realizar una primera exploración del dataset.

### 2.1 Importar las bibliotecas

In [1]:
import pandas as pd
import numpy as np

### 2.2 Cargar el archivo CSV

In [2]:
df = pd.read_csv("../data/raw/annual-population-growth/annual-population-growth.csv")

### 2.3 Explorar el DataFrame

**Primeras filas**

In [3]:
df.head()

,Entity,Code,Year,Annual change in population,Annual population change (Projected)
0,Afghanistan,AFG,1951,103163.0,NaN
1,Afghanistan,AFG,1952,108441.0,NaN
2,Afghanistan,AFG,1953,108919.0,NaN
3,Afghanistan,AFG,1954,111251.0,NaN
4,Afghanistan,AFG,1955,119027.0,NaN


**Dimensiones**

In [4]:
df.shape

(38400, 5)

**Nombres de columnas**

In [5]:
df.columns.tolist()

['Entity',
 'Code',
 'Year',
 'Annual change in population',
 'Annual population change (Projected)']

**Tipos de datos**

In [6]:
df.dtypes

Entity                                      str
Code                                        str
Year                                      int64
Annual change in population             float64
Annual population change (Projected)    float64
dtype: object

**Estadísticas descriptivas**

In [7]:
df.describe()

,Year,Annual change in population,Annual population change (Projected)
count,38400.000000,1.868800e+04,1.971200e+04
mean,2025.500000,2.063019e+06,8.375481e+05
std,43.300872,9.440859e+06,5.816745e+06
min,1951.000000,-3.315935e+06,-2.291035e+07
25%,1988.000000,1.781000e+03,-9.742750e+03
50%,2025.500000,4.689800e+04,9.350000e+01
75%,2063.000000,3.021128e+05,1.411160e+05
max,2100.000000,9.337137e+07,7.206736e+07


**Cantidad de valores faltantes**

In [8]:
df.isnull().sum()

Entity                                      0
Code                                     1200
Year                                        0
Annual change in population             19712
Annual population change (Projected)    18688
dtype: int64

**Cantidad de entidades**

In [9]:
df["Entity"].nunique()

256

### 2.4 Selección de datos con `loc` e `iloc`

Ejemplos adicionales de selección de filas y columnas por etiqueta (`loc`) y por posición (`iloc`).

In [10]:
df.loc[10000]

Entity                                  Equatorial Guinea
Code                                                  GNQ
Year                                                 2051
Annual change in population                           NaN
Annual population change (Projected)              46407.0
Name: 10000, dtype: object

In [11]:
df.loc[:,"Entity"]

0        Afghanistan
1        Afghanistan
2        Afghanistan
3        Afghanistan
4        Afghanistan
            ...     
38395       Zimbabwe
38396       Zimbabwe
38397       Zimbabwe
38398       Zimbabwe
38399       Zimbabwe
Name: Entity, Length: 38400, dtype: str

In [12]:
df.iloc[30:35]

,Entity,Code,Year,Annual change in population,Annual population change (Projected)
30,Afghanistan,AFG,1981,-1231727.0,NaN
31,Afghanistan,AFG,1982,-946206.0,NaN
32,Afghanistan,AFG,1983,-73395.0,NaN
33,Afghanistan,AFG,1984,272237.0,NaN
34,Afghanistan,AFG,1985,236633.0,NaN


In [13]:
df.loc[:,["Entity","Year", "Annual change in population"]]

,Entity,Year,Annual change in population
0,Afghanistan,1951,103163.0
1,Afghanistan,1952,108441.0
2,Afghanistan,1953,108919.0
3,Afghanistan,1954,111251.0
4,Afghanistan,1955,119027.0
...,...,...,...
38395,Zimbabwe,2096,NaN
38396,Zimbabwe,2097,NaN
38397,Zimbabwe,2098,NaN
38398,Zimbabwe,2099,NaN


In [14]:
df.loc[10000,["Entity","Year", "Annual change in population"]]

Entity                         Equatorial Guinea
Year                                        2051
Annual change in population                  NaN
Name: 10000, dtype: object

In [15]:
df.loc[(df["Entity"] == "Mexico") & (df["Year"] == 2020)]

,Entity,Code,Year,Annual change in population,Annual population change (Projected)
22269,Mexico,MEX,2020,1036075.0,NaN


In [16]:
df.iloc[0:5]

,Entity,Code,Year,Annual change in population,Annual population change (Projected)
0,Afghanistan,AFG,1951,103163.0,NaN
1,Afghanistan,AFG,1952,108441.0,NaN
2,Afghanistan,AFG,1953,108919.0,NaN
3,Afghanistan,AFG,1954,111251.0,NaN
4,Afghanistan,AFG,1955,119027.0,NaN


In [17]:
df.iloc[:, [0,2,3]]

,Entity,Year,Annual change in population
0,Afghanistan,1951,103163.0
1,Afghanistan,1952,108441.0
2,Afghanistan,1953,108919.0
3,Afghanistan,1954,111251.0
4,Afghanistan,1955,119027.0
...,...,...,...
38395,Zimbabwe,2096,NaN
38396,Zimbabwe,2097,NaN
38397,Zimbabwe,2098,NaN
38398,Zimbabwe,2099,NaN


In [18]:
df.iloc[0:5, 0:3]

,Entity,Code,Year
0,Afghanistan,AFG,1951
1,Afghanistan,AFG,1952
2,Afghanistan,AFG,1953
3,Afghanistan,AFG,1954
4,Afghanistan,AFG,1955


### 2.5 Registro de un país que empieza con "S" (2005–2026)

Se filtra el DataFrame para un país cuyo nombre inicia con "S" (**Samoa**), en el rango de años 2005 a 2026. Como este rango cruza el límite entre datos históricos (hasta 2023) y proyectados (desde 2024), se muestran ambas columnas de cambio poblacional.

In [19]:
log_pais_s = df.loc[(df["Entity"] == "Samoa") & (df["Year"] >= 2005) & (df["Year"] <= 2026)]
log_pais_s

,Entity,Code,Year,Annual change in population,Annual population change (Projected)
29904,Samoa,WSM,2005,728.0,NaN
29905,Samoa,WSM,2006,731.0,NaN
29906,Samoa,WSM,2007,1042.0,NaN
29907,Samoa,WSM,2008,1351.0,NaN
29908,Samoa,WSM,2009,1413.0,NaN
29909,Samoa,WSM,2010,1505.0,NaN
29910,Samoa,WSM,2011,1691.0,NaN
29911,Samoa,WSM,2012,1792.0,NaN
29912,Samoa,WSM,2013,1766.0,NaN
29913,Samoa,WSM,2014,1767.0,NaN


In [20]:
log_pais_s.mean(numeric_only=True)

Year                                    2015.500000
Annual change in population             1600.421053
Annual population change (Projected)    1289.333333
dtype: float64

In [21]:
log_pais_s.median(numeric_only=True)

Year                                    2015.5
Annual change in population             1715.0
Annual population change (Projected)    1284.0
dtype: float64

In [22]:
log_pais_s.var(numeric_only=True)

Year                                        42.166667
Annual change in population             182090.701754
Annual population change (Projected)      4377.333333
dtype: float64

In [23]:
log_pais_s.std(numeric_only=True)

Year                                      6.493587
Annual change in population             426.720871
Annual population change (Projected)     66.161419
dtype: float64

In [24]:
log_pais_s.min(numeric_only=True)

Year                                    2005.0
Annual change in population              728.0
Annual population change (Projected)    1226.0
dtype: float64

In [25]:
log_pais_s.max(numeric_only=True)

Year                                    2026.0
Annual change in population             2199.0
Annual population change (Projected)    1358.0
dtype: float64

### Preguntas

1. **¿Cuántas filas y columnas tiene el dataset?**
   El dataset tiene 38,400 filas y 5 columnas.

2. **¿Qué tipo de datos contiene la variable `Year`?**
   Es de tipo entero (`int64`).

3. **¿Existen valores faltantes?**
   Sí. La columna `Code` tiene 1,200 valores faltantes, `Annual change in population` tiene 19,712, y `Annual population change (Projected)` tiene 18,688. Esto se debe a que el dataset combina datos históricos (1951–2023) y proyectados (2024–2100) en columnas separadas, por lo que cada fila solo tiene valor en una de las dos columnas de cambio poblacional, nunca en ambas.

4. **¿Cuántas entidades diferentes contiene el dataset?**
   Contiene 256 entidades (países, regiones y agrupaciones como "World" o "Africa (UN)").

## Actividad 3 — Estadística descriptiva

**Propósito:** aplicar medidas estadísticas básicas a una variable poblacional.

Se selecciona el año **2020** (dentro del periodo histórico, 1951–2023) y se filtran los datos correspondientes. Sobre la variable `Annual change in population` se calculan media, mediana, varianza, desviación estándar, mínimo y máximo.

In [26]:
df_2020 = df[df["Year"] == 2020]
cambio_poblacional = df_2020["Annual change in population"].dropna()
cambio_poblacional.head()

69      1212852.0
219    32815675.0
369      -13061.0
519      747543.0
669        -451.0
Name: Annual change in population, dtype: float64

In [27]:
media = cambio_poblacional.mean()
mediana = cambio_poblacional.median()
varianza = cambio_poblacional.var()
desviacion_estandar = cambio_poblacional.std()
minimo = cambio_poblacional.min()
maximo = cambio_poblacional.max()

print("Media:", media)
print("Mediana:", mediana)
print("Varianza:", varianza)
print("Desviación estándar:", desviacion_estandar)
print("Mínimo:", minimo)
print("Máximo:", maximo)

Media: 2124908.84765625
Mediana: 34293.5
Varianza: 90687335031103.64
Desviación estándar: 9522989.815761836
Mínimo: -494016.0
Máximo: 75707590.0


**Entidades con el valor mínimo y máximo**

In [28]:
entidad_min = df_2020.loc[cambio_poblacional.idxmin(), "Entity"]
entidad_max = df_2020.loc[cambio_poblacional.idxmax(), "Entity"]

print("Entidad con el valor mínimo:", entidad_min, "->", minimo)
print("Entidad con el valor máximo:", entidad_max, "->", maximo)

Entidad con el valor mínimo: Venezuela -> -494016.0
Entidad con el valor máximo: World -> 75707590.0


### Preguntas

1. **¿Cuál es el valor de la media?**
   Aproximadamente 2,124,908.85 personas.

2. **¿Cuál es el valor de la mediana?**
   34,293.5 personas.

3. **¿Cuál es el valor de la desviación estándar?**
   Aproximadamente 9,522,989.82 personas.

4. **¿Qué entidad presenta el valor mínimo y cuál el máximo?**
   El valor mínimo (-494,016) corresponde a Venezuela. El valor máximo (75,707,590) corresponde a "World", que es una entidad agregada (el total mundial) y no un país individual; entre los países, el mayor crecimiento en 2020 lo tuvo India.

## Actividad 4 — Comprobación del entorno

**Propósito:** comprobar que el notebook está utilizando correctamente las herramientas configuradas para la práctica.

In [29]:
import sys

print("Python:", sys.version)
print("Pandas:", pd.__version__)
print("NumPy:", np.__version__)

Python: 3.14.7 (tags/v3.14.7:823f032, Aug  5 2026, 10:51:32) [MSC v.1944 64 bit (AMD64)]
Pandas: 3.0.5
NumPy: 2.5.2


## Resultados

Este notebook mostró:

- La información básica de la fuente de los datos y su cita correspondiente (Actividad 1).
- La carga del DataFrame `annual-population-growth.csv` y su estructura: dimensiones, columnas, tipos de datos y valores faltantes (Actividad 2).
- Los estadísticos descriptivos (media, mediana, varianza, desviación estándar, mínimo y máximo) de la variable `Annual change in population` para el año 2020 (Actividad 3).
- Las versiones de Python, Pandas y NumPy utilizadas, confirmando que el entorno configurado para el curso funciona correctamente (Actividad 4).

No se incluyen gráficas, conclusiones ni análisis de proyecciones o temporales, ya que estos temas se abordarán en prácticas posteriores.